telekom-workflow-engine/src/main/java/ee/telekom/workflow/executor/marshall/Marshaller.java

```diff
@@ -32,14 +32,15 @@
 public class Marshaller{
 
     public static void marshall( GraphInstance instance, WorkflowInstance woin, List<WorkItem> woits, WorkflowInstanceStatus completeStatus ){
+        boolean isCompleted = instance.isCompleted();
         woin.setRefNum( instance.getExternalId() );
         woin.setWorkflowName( instance.getGraph().getName() );
         woin.setWorkflowVersion( instance.getGraph().getVersion() );
         woin.setAttributes( serializeAttributes( instance.getEnvironment().getAttributesAsMap() ) );
-        woin.setHistory( instance.getGraph().getKeepHistory() ? instance.getHistory()
-                : HistoryUtil.deleteHistory( instance.getHistory() ) );
-        woin.setState( serializeTokens( instance.getTokens(), instance.getGraph().getKeepHistory() ) );
-        woin.setStatus( instance.isCompleted() ? completeStatus : WorkflowInstanceStatus.EXECUTING );
+        woin.setHistory( !instance.getGraph().getKeepHistory() && !isCompleted ? HistoryUtil.deleteHistory( instance.getHistory() ) 
```

Comment:
Please wrap these two conditions `!instance.getGraph().getKeepHistory() && !isCompleted` into parenthesis, so that it's chrystal clear, how the `?:` statement flows.

Reply:
I wrapped these two conditions into parenthesis. Now this statement should be clearer.